rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java

```diff
@@ -91,6 +102,26 @@ protected void prePopulateRDTPatientFields(Patient patient, JSONObject field) th
         } else if (PATIENT_AGE.equals(key)) {
             field.put(JsonFormUtils.VALUE, patient.getAge());
         }
+
+        // pre-populate the patient detail name
+        if (CovidConstants.FormFields.PATIENT_INFO_NAME.equals(field.getString(KEY))) {
+            JSONArray options = field.getJSONArray(JsonFormConstants.OPTIONS_FIELD_NAME);
+            if (options.length() > 0) {
+                JSONObject nameObject = options.getJSONObject(0);
+                String nameValue = nameObject.getString(JsonFormConstants.TEXT);
+                nameObject.put(JsonFormConstants.TEXT, nameValue + patient.getPatientName());
+            }
+        }
+
+        // pre-populate the patient detail dob
+        if (CovidConstants.FormFields.PATIENT_INFO_DOB.equals(field.getString(KEY))) {
+            JSONArray options = field.getJSONArray(JsonFormConstants.OPTIONS_FIELD_NAME);
+            if (options.length() > 0) {
+                JSONObject dobObject = options.getJSONObject(0);
+                String dobValue = dobObject.getString(JsonFormConstants.TEXT);
+                dobObject.put(JsonFormConstants.TEXT, dobValue + patient.getDob());
+            }
+        }
```

Comment:
These three blocks have similar functionality. Could this be factored out / generalized into a function that can be called three times in place of these blocks?